Webservice structural variant

dionnezaal opened the pull request:
Moved old legacy structural variant API to new API. 
API contains one endpoint:

```
POST /structuralvariant/fetch

{
  "geneticProfileStableIds": [
    "string"
  ],
  "hugoGeneSymbols": [
    "string"
  ],
  "sampleIdentifiers": [
    {
      "sampleId": "string",
      "studyId": "string"
    }
  ]
}
```

`geneticProfileStableIds` and `hugoGeneSymbols` are required.
No different projections, this means it will output the whole structuralVariant model for each structuralVariant with has provided `geneticProfileStableIds` and `hugoGeneSymbols`.

Commit:
Remove old structural variant

Commit:
Add documentation for structural variation

Commit:
Add importer for structural variation data type.

Commit:
Add custom driver annotation columns

Commit:
small changes to data model

pieterlukasse committed:
added optional fieldNames for tables with AI

Commit:
Work in progress on structural_variant

dionnezaal committed:
Start working on new API structural variant

Commit:
fix mysql bulkloader for structural_variant

Commit:
styling

dionnezaal committed:
API for structural variants with one POST endpoint

Commit:
fix unit test

Commit:
Fix unit tests

dionnezaal committed:
Added my batis unit test

dionnezaal committed:
Fixed AssertTrue boolean method problem

dionnezaal committed:
Fixed mybatis unit test

pieterlukasse commented on persistence/persistence-mybatis/src/main/resources/org/cbioportal/persistence/mybatis/StructuralVariantMapper.xml:
```diff
@@ -0,0 +1,90 @@
+        structural_variant.GENETIC_PROFILE_ID AS "${prefix}geneticProfileId",
+        structural_variant.INTERNAL_ID AS "${prefix}structuralVariantId",
+        structural_variant.SAMPLE_ID AS "${prefix}sampleIdInternal",
+        <if test="studyIds.size() > 0 and sampleIds.size() > 0">
```
please remove this. Sample id should always be in response.

pieterlukasse commented on persistence/persistence-mybatis/src/main/resources/org/cbioportal/persistence/mybatis/StructuralVariantMapper.xml:
```diff
@@ -0,0 +1,90 @@
+        JOIN genetic_profile ON structural_variant.GENETIC_PROFILE_ID = genetic_profile.GENETIC_PROFILE_ID
+        JOIN gene gene1 ON structural_variant.SITE1_ENTREZ_GENE_ID = gene1.ENTREZ_GENE_ID
+        JOIN gene gene2 ON structural_variant.SITE2_ENTREZ_GENE_ID = gene2.ENTREZ_GENE_ID
+        <if test="studyIds.size() > 0 and sampleIds.size() > 0">
```
please remove this `if`. The join should always be made since sample id and cancer study ids should be in the response (e.g. in the scenario where one queries without specifying sample ids or cancer study, we return for all samples and studies right? )

pieterlukasse commented on persistence/persistence-mybatis/src/main/resources/org/cbioportal/persistence/mybatis/StructuralVariantMapper.xml:
```diff
@@ -0,0 +1,90 @@
+                (#{studyIds[${i}]}, #{sampleIds[${i}]}) 
+            </foreach>
+        </if>
+        </where>
```
please also add an `ORDER BY` clause to your query. This can be `ORDER BY gene1.HUGO_GENE_SYMBOL, gene2.HUGO_GENE_SYMBOL`

pieterlukasse commented on persistence/persistence-mybatis/src/test/java/org/cbioportal/persistence/mybatis/StructuralVariantMyBatisRepositoryTest.java:
```diff
@@ -0,0 +1,147 @@
+                structuralVariantMyBatisRepository.fetchStructuralVariants(geneticProfileStableIds, 
+                        hugoGeneSymbols, studyIds, sampleIds);
+
+        Assert.assertEquals(2,  result.size());
```
ideally this will be different from previous test, so that you can check if the extra parameter is actually filtering anything.

pieterlukasse reviewed (changes requested):
Nicely done! 
I did add some minor change/improvement requests.

dionnezaal committed:
Removed if statement in SQL query to always show sample identifier

dionnezaal committed:
Added ORDER BY hugo gene symbols to SQL query

dionnezaal committed:
Adapted testSQL to be able to test select of samples, adapted result of mybatis unit test

dionnezaal commented:
@pieterlukasse I have removed the if statements for the `SAMPLE_ID`, now it will always be shown in the result. I have also added an ORDER BY statement in the SQL query. 

I have adapted the test database file to include two samples in the structural variant table and used that for the mybatisrepository unit test in case `sampleIdentifiers` are supplied to test if only the structural variant from the sample selected is shown.

dionnezaal committed:
Fix for date time format to prevent unit test fail

dionnezaal committed:
Removed unused import statement in service layer

dionnezaal committed:
Added ServiceImpl and Controller Unit test for structural variant API

dionnezaal commented:
Have added unit tests for controller and serviceImpl to prevent the web layer unit tests from failing.

dionnezaal commented:
Closing this PR because we will continue to work on rebased version PR #25.
